xapian-letor/tests/Makefile.am

```diff
@@ -81,14 +82,46 @@ EXTRA_DIST = collate-test generate-api_generated api_all.h api_collated.h \
 EXTRA_DIST +=\
 	runtest.in \
 	valgrind.supp \
-	testdata/apitest_rset.txt \
-	testdata/apitest_phrase.txt \
 	testdata/apitest_allterms4.txt \
+	testdata/apitest_phrase.txt \
+	testdata/apitest_rset.txt \
+	testdata/empty_file.txt \
 	testdata/qrel.txt \
+	testdata/qrelone.txt \
+	testdata/qrelthree_correct.txt \
 	testdata/query.txt \
+	testdata/queryone.txt \
+	testdata/querythree.txt \
 	testdata/score_qrel.txt \
 	testdata/training_data.txt \
-	testdata/training_data_ndcg.txt
+	testdata/training_data_ndcg.txt \
+	testdata/training_data_one_document.txt \
+	testdata/training_data_three_correct.txt
+
+DISTCLEANFILES +=\
+	err_output_listmle_1.txt \
+	err_output_listmle_2.txt \
+	err_output_listmle_3.txt \
+	err_output_ListNet_1.txt \
+	err_output_ListNet_2.txt \
+	err_output_ListNet_3.txt \
+	err_output_svm_1.txt \
+	err_output_svm_2.txt \
+	err_output_svm_3.txt \
+	ndcg_output_listmle_1.txt \
+	ndcg_output_listmle_2.txt \
+	ndcg_output_listmle_3.txt \
+	ndcg_output_ListNet_1.txt \
+	ndcg_output_ListNet_2.txt \
+	ndcg_output_ListNet_3.txt \
+	ndcg_output_svm_1.txt \
+	ndcg_output_svm_2.txt \
+	ndcg_output_svm_3.txt \
+	ndcg_score_test.txt \
+	training_output_data_one_doc.txt \
+	training_output_empty.txt \
+	training_output_three_correct.txt \
+	training_output1.txt
```

Comment:
These are all output files produced and used within a single testcase aren't they?

If so, I think it would be good for these testcases to at least check that the expected output file is created (ideally the content would be checked too in some appropriate way) and then remove it - e.g. see how I did in 4a27dcf65fcfcc670768dee90396dfca17ed304a for `ndcg_score_test.txt`.  If such a testcase fails the output file will be left behind, so one can still check it in the case where you'd want to.

It's still good to be cleaning up these files here (though I think they want to be in `CLEANFILES` so they are removed on `make clean` not just `make distclean`) in case they're left behind by a failing testcase or by an interrupted testsuite run.

Reply:
fixed in a fixup commit.